lib/src/download_utils.dart

```diff
@@ -153,6 +153,10 @@ class UrlChecker {
     return _internalHosts.every((p) => p.allMatches(uri.host).isEmpty);
   }
 
+  Future<bool> exists(Uri uri) async {
```

Comment:
Do you want to subclass and inject a custom `UrlChecker`?

If so, I think documenting all members would be a good idea.

But if you want some caching, why not just cache the last 1000 URLs we looked using an in-memory dictionary? Using a shared cache for this feels like overkill..

Reply:
Good point, updated it, PTAL.

However, I'd like to keep the option open to implement a redis-based cache too, so updated the API accordingly...